src/main/java/com/joss/conductor/mobile/Locomotive.java

```diff
@@ -572,19 +572,23 @@ private Locomotive performCornerSwipe(ScreenCorner corner, SwipeElementDirection
     }
 
     public WebElement swipeTo(SwipeElementDirection s, By by, int attempts) {
-        int i;
-
-        if (isPresentWait(by)) {
-            return driver.findElement(by);
-        } else {
-            for (i = 0; i < attempts; i++) {
-                swipeCenterLong(s);
-                if (isPresent(by)) {
-                    return driver.findElement(by);
+        WebElement element;
+        for (int i = 0; i < attempts; i++) {
+            swipeCenterLong(s);
+            try {
+                element = driver.findElement(by);
+                // element was found, check for visibility
+                if (element.isDisplayed()) {
+                    // element is in view, exit the loop
+                    return element;
                 }
+                // element was not visible, continue scrolling
+            } catch (org.openqa.selenium.NoSuchElementException e) {
+                // element could not be found, continue scrolling
             }
         }
-        Logger.error("WARN: Element" + by.toString() + "does not exist!");
+        // element could not be found or was not visible, return null
+        System.err.println("WARN: Element" + by.toString() + "does not exist!");
```

Comment:
Let's use the logging library so we can turn it off and on. Also the `WARN:` should be generated by the correct level, so `Logger.warn("Element" + by.toString() + "does not exist!");`